src/tabbar.cpp

```diff
@@ -151,20 +122,31 @@ void TabBar::moveToPreviousTab()
     setCurrentIndex(index <= 0 ? realTabCount() - 1 : index - 1);
 }
 
-ZimView* TabBar::createNewTab(bool setCurrent, bool adjacentToCurrentTab)
+void TabBar::setCloseTabButton(int index)
 {
-    auto tab = new ZimView(this, this);
-    int index;
-    if(adjacentToCurrentTab) {
-        index = currentIndex() + 1;
-    } else {
-        index = realTabCount(); // for New Tab Button
-    }
-    mp_stackedWidget->insertWidget(index, tab);
-    index = insertTab(index, "");
+    Q_ASSERT(index > 0 && index < realTabCount());
+
     QToolButton *tb = new QToolButton(this);
-    tb->setDefaultAction(KiwixApp::instance()->getAction(KiwixApp::CloseTabAction));
+    QAction *a = new QAction(QIcon(":/icons/close.svg"), gt("close-tab"), tb);
```

Comment:
The button originally had a tooltip about the keyboard shortcut. Should we include it here as well?

Reply:
Done